locks.go

```diff
@@ -143,3 +148,38 @@ func (l *LockRedis) Unlock(ctx context.Context) error {
 	}
 	return nil
 }
+
+// LockMemcached is a wrapper around github.com/alessandro-c/gomemcached-lock that implements the DistLocker interface.
+// It is caller's responsibility to make sure the uniqueness of mutexName, and not to use the same key in multiple
+// Memcached-based implementations
+type LockMemcached struct {
+	locker    *lock.Locker
+	mutexName string
+	backoff   backoff.BackOff
+}
+
+// NewLockMemcached creates a new instance of LockMemcached.
+// b the backoff policy for retrying an operation. Default to retry every 100ms for 100 times (10 seconds).
+func NewLockMemcached(client *memcache.Client, mutexName string, b backoff.BackOff) *LockMemcached {
```

Comment:
I think we can remove the `b backoff.BackOff` argument from this function definition, use some default value but provide a way of adjusting this value in go-style manner like this:


```golang
func (c *LockMemcached) WithLockAcquireBackoff(b backoff.BackOff) *LockMemcached {
    c.backoff = b
    return c
}
```

Reply:
No problem.